pkg/native/serializers/serializer_spdx23.go

```diff
@@ -21,8 +23,19 @@ var _ native.Serializer = &SPDX23{}
 
 type SPDX23 struct{}
 
-type SPDX3Options struct {
-	Indent int
+type SPDX23Options struct {
+	// FailOnInvalidDocIdFragment makes the serializer return an error if the
+	// document ID has a fragment but it is not set to "SPDXRef-Document".
+	FailOnInvalidDocIdFragment bool
+
+	// GenerateDocumentId causes the serializer to generate a non-deterministic
+	// document ID when a protobom doesn't have one defined.
+	GenerateDocumentId bool
```

Comment:
just to follow golang code practices this can be called

```suggestion
	GenerateDocumentID bool
```